src/convertToObject.js

```diff
@@ -6,7 +6,44 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitArray = sourceString.trim().split(';');
+  const object = {};
+
+  for (const item of splitArray) {
+    if (item.trim().length > 0) {
+      const resultItemSplit = item.split(':');
+      const key = resultItemSplit[0].trim();
+      const value = resultItemSplit[1].trim();
+
+      object[key] = value;
+    }
+  }
+
+  return object;
```

Comment:
The function `convertToObject` uses a for loop to iterate over the array. Consider using array iteration methods like `map` and `filter` to transform the string into an object as per the checklist.